fragments/infra-boot.sh

```diff
@@ -50,6 +50,10 @@ function notify_failure() {
 # 
 # --- DNS functions ----------------------------------------------------------
 #
+function save_resolver_file() {
+    cp /etc/resolv.conf /etc/resolv.conf.local
+}
+
```

Comment:
only nit (not a blocker for merging this patchset): these one-liner functions (where the line itself is very simple anyway) and these lines are used on single place  (so there is no re-use of them) is too much over-engineering and decreases readability of the code (albeit the intention I guess was opposite) because you have to skip here (function body) and there (call of the function)